packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -561,10 +561,6 @@ function createDOMResponderEvent(
     target: nativeEventTarget,
     type: topLevelType,
   };
-  if (__DEV__) {
-    Object.freeze(responderEvent);
```

Comment:
We don't want to freeze the object anymore, as we might plan on recycling/mutating the object in the future for perf reasons.